frontend/src/App.jsx

```diff
@@ -1,19 +1,74 @@
-import { BrowserRouter, Routes, Route } from "react-router-dom";
+import { BrowserRouter, Routes, Route, Navigate } from "react-router-dom";
 import "./App.css";
 import Login from "./Login";
 import SignUp from "./SignUp";
 import Home from "./Home";
-import { useState } from "react";
+import { useState, useEffect } from "react";
+import { BASE_URL } from "./utils/reused";
 
 const App = () => {
+  const [isLoggedIn, setLoggedIn] = useState(false);
+
+  const attemptLogin = async () => {
+    try {
+      const response = await fetch(`${BASE_URL}/auth/me`, {
+        method: "GET",
+        credentials: "include",
+      });
+      if (response.ok) {
+        setLoggedIn(true);
+      }
+    } catch {
+      return;
+    }
+  };
+
+  attemptLogin();
+
   return (
-    <BrowserRouter>
-      <Routes>
-        <Route path="/" element={<Login />} />
-        <Route path="/home" element={<Home />} />
-        <Route path="/signup" element={<SignUp />} />
-      </Routes>
-    </BrowserRouter>
+    <>
+      <BrowserRouter>
+        <Routes>
+          <Route
+            path="/"
+            element={
+              isLoggedIn === true ? (
+                <Navigate to="/home" />
+              ) : (
+                <Navigate to="/login" />
+              )
+            }
+          />
+          <Route
+            path="/login"
+            element={
+              isLoggedIn === true ? (
+                <Navigate to="/home" />
+              ) : (
+                <Login attemptLogin={attemptLogin} />
+              )
+            }
+          />
+          <Route
+            path="/home"
+            element={isLoggedIn === true ? <Home /> : <Navigate to="/login" />}
```

Comment:
I'm not an expert at React Router but you should be able to refactor this into a new sub-component

e.g.
```
<Route
 path="/home"
 element={<LoggedInElement isLoggedIn={isLoggedIn}><Home /></LoggedInElement>}
/>
```

```
const LoggedInElement = ({ isLoggedIn, children }) => {
 if (isLoggedIn) {
   return children;
 }
 return <Navigate to="/login" />;
}
```

Then you can reuse this at line 32-41 as well.

I think you could even try to refactor it further by create a wrapping component that return <Route>